### PR TITLE
512 vital connections UI

### PR DESCRIPTION
**Pull request by GennadiyShpak**

### CODE REVIEW

File: src/views/connections/types/azureblob.tsx [hunk 1]
```diff
@@ -6,17 +6,21 @@ import {ConnectionSpecPlugin, ConnectionTypes} from "./index";
...
-        {name: "Shared Access Signature URL", elem: hidingSequence}
+        {name: "Shared Access Signature URL", elem: hidingSequence},
+        {name: "Vital", elem: conn.spec?.vital? "true": "false"}
     ];
 }
 
 const Schema = {
     uri: Yup.string().trim().required('Shared Access Signature URL is a required field'),
     keySecret: Yup.string().trim().required('Shared Access Token is a required field'),
+    vital: Yup.string().trim(),
```
Comment (lev-titov) [inline src/views/connections/types/azureblob.tsx]: i guess its boolean now, not string

File: src/views/connections/types/docker.tsx [hunk 2]
```diff
@@ -7,19 +7,22 @@ import {ConnectionSpecPlugin, ConnectionTypes} from "./index";
...
+        {name: "Vital", elem: conn.spec?.vital? "true": "false"}
     ];
 }
 
 export const Schema = {
     uri: Yup.string().trim().required('URI is a required field'),
     username: Yup.string().trim().required('Username is a required field'),
-    password: Yup.string().trim().required('Password is a required field')
+    password: Yup.string().trim().required('Password is a required field'),
+    vital: Yup.string().trim(),
```
Comment (lev-titov) [inline src/views/connections/types/docker.tsx]: boolean

File: src/views/connections/types/ecr.tsx [hunk 3]
```diff
@@ -22,6 +24,7 @@ export const Schema = {
     keySecret: Yup.string().trim().required('Access Key Secret is a required field'),
     keyID: Yup.string().trim().required('Access Key ID is a required field'),
     region: Yup.string().trim(),
+    vital: Yup.string().trim(),
```
Comment (lev-titov) [inline src/views/connections/types/ecr.tsx]: boolean

File: src/views/connections/types/gcs.tsx [hunk 4]
```diff
@@ -7,21 +7,23 @@ import {ConnectionSpecPlugin, ConnectionTypes} from "./index";
...
+        {name: "Service Account Key", elem: hidingSequence},
+        {name: "Vital", elem: conn.spec?.vital? "true": "false"}
     ];
 }
 
 export const Schema = {
     uri: Yup.string().trim().required('URI is a required field'),
     region: Yup.string().trim().required('Project is a required field'),
     keySecret: Yup.string().trim().required('Service account key is a required field'),
+    vital: Yup.string().trim(),
```
Comment (lev-titov) [inline src/views/connections/types/gcs.tsx]: boolean

File: src/views/connections/types/git.tsx [hunk 5]
```diff
@@ -8,19 +8,22 @@ import {OdahuTextField} from "../../../components/OdahuTextField";
...
+        {name: "SSH private key", elem: hidingSequence},
+        {name: "Vital", elem: conn.spec?.vital? "true": "false"}
     ];
 }
 
 export const Schema = {
     uri: Yup.string().trim().required('GIT URL is a required field'),
     keySecret: Yup.string().trim().required('SSH private key is a required field'),
     reference: Yup.string().trim(),
+    vital: Yup.string().trim(),
```
Comment (lev-titov) [inline src/views/connections/types/git.tsx]: boolean

File: src/views/connections/types/s3.tsx [hunk 6]
```diff
@@ -23,6 +25,7 @@ export const Schema = {
     keySecret: Yup.string().trim().required('Access Key Secret is a required field'),
     keyID: Yup.string().trim().required('Access Key ID is a required field'),
     region: Yup.string().trim().required('Region is a required field'),
+    vital: Yup.string().trim(),
```
Comment (lev-titov) [inline src/views/connections/types/s3.tsx]: boolean

File: src/components/ConnectionCheckbox.tsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                    formik.setFieldValue(name, value);
+                }}
+            ></Checkbox>
+        </label>
+        <p className={(classes.paragraph, classes.description)}>
+            <span className={classes.helperText}>{description}</span>
+        </p>
+    </div>
+    )
+};
```
Comment (lev-titov) [inline src/components/ConnectionCheckbox.tsx]: why is it called connection checkbox?
i assume we would like to use checkboxes in some other places later, not only for connections

**Review by lev-titov — CHANGES_REQUESTED** [review]

File: src/views/connections/types/docker.tsx (hunk 2 above)
Comment (GennadiyShpak) [reply src/views/connections/types/docker.tsx]: fixed

File: src/views/connections/types/azureblob.tsx (hunk 1 above)
Comment (GennadiyShpak) [reply src/views/connections/types/azureblob.tsx]: fixed

File: src/views/connections/types/ecr.tsx (hunk 3 above)
Comment (GennadiyShpak) [reply src/views/connections/types/ecr.tsx]: fixed

File: src/views/connections/types/gcs.tsx (hunk 4 above)
Comment (GennadiyShpak) [reply src/views/connections/types/gcs.tsx]: fixed

File: src/views/connections/types/git.tsx (hunk 5 above)
Comment (GennadiyShpak) [reply src/views/connections/types/git.tsx]: fixed

File: src/views/connections/types/s3.tsx (hunk 6 above)
Comment (GennadiyShpak) [reply src/views/connections/types/s3.tsx]: fixed

File: src/components/ConnectionCheckbox.tsx (hunk 7 above)
Comment (GennadiyShpak) [reply src/components/ConnectionCheckbox.tsx]: Discussed offline! i changed the name of the component

**Comment by owlleg6** [comment]
linked to https://github.com/odahu/odahu-flow/pull/650

**Review by lev-titov — APPROVED** [review]

**Review by owlleg6 — APPROVED** [review]

**Pull request merged**